src/V3Width.cpp

```diff
@@ -3298,6 +3298,26 @@ class WidthVisitor final : public AstNVisitor {
                     relinkHandle.relink(patp);
                 }
             }
+
+            std::vector<std::pair<AstNodeDType*, AstPatMember*>> type_defaults;
+            for (AstPatMember* patp = VN_CAST(nodep->itemsp(), PatMember); patp;
+                 ) {
+                if (patp->isTypedDefault()) {
+                    AstNodeDType* type = VN_CAST(patp->keyp(), NodeDType);
+                    auto matching_type_it
+                        = find_if(type_defaults.begin(), type_defaults.end(),
+                                  [&type](auto x) { return type->similarDType(x.first); });
+                    if (matching_type_it == type_defaults.end()) {
```

Comment:
Does the test cover this case?

Reply:
I think that yes. [First case](https://github.com/Aj0SK/verilator/blob/2ccfc29e323d0c157ec9c0557a404c4fd524dfd2/test_regress/t/t_struct_assignment.v#L21) and [second case](https://github.com/Aj0SK/verilator/blob/2ccfc29e323d0c157ec9c0557a404c4fd524dfd2/test_regress/t/t_struct_assignment.v#L22).